static/app/components/searchSyntax/renderer.tsx

```diff
@@ -92,6 +106,26 @@ const FilterToken = ({
   const showInvalid = hasLeft && !!filter.invalid;
   const showTooltip = showInvalid && isActive;
 
+  const reduceMotion = useReducedMotion();
+
+  // Trigger the shakeAnimation when showInvalid is set to true. We reset the
+  // animation by clearing the style, set it to running, and re-applying the
+  // animation
+  useEffect(() => {
+    if (!filterElementRef.current || !showInvalid || reduceMotion) {
+      return;
+    }
+
+    const style = filterElementRef.current.style;
+
+    style.animation = 'none';
+    void filterElementRef.current.offsetTop;
+
+    window.requestAnimationFrame(
+      () => (style.animation = `${shakeAnimation.name} 300ms`)
```

Comment:
I was hoping I could let emotion take care of applying and removing the animation, but unfortunately it seems like it does it more often than I would like it to, so we have to manually handle this here